client/main.py

```diff
@@ -28,6 +30,13 @@
         should_stop = GameSession.run_session(zmq_ctx, game_win, player_uuid, address)
 
```

Comment:
This entire segment needs to be reworked. The game loop was originally written without an end screen in mind, so we shouldn't take debt and shoehorn this bit into here. Here are some things to take into consideration:

 - The flow of the game states would be something like login -> gamesession -> end. From end, you can either go back to login, or directly into gamesession
 - From the end screen, the player should also be able to exit out of the game. When the game was exited from the gamesession, it should properly disconnect the client and exit the program without showing the end screen. At the moment, if I press x during the gamesession, it takes me to the end screen.
 - So if the end screen can branch out to three different things (login, gamesession, exit game), we need to have it return three possible values.

I'm thinking of something like this (omitting details on everything else):

```py
rejoin = False
while True:
    if not rejoin:
        address = login()
    should_stop = gamesession(address)
    if not should_stop:
        rejoin = endscreen() # the 3 possible values here are true false or none
        if rejoin is not None:
            continue
    disconnect()
```

Reply:
try to step through the pseudocode here and imagine what the player could do in each state, to convince yourself that this works (or doesn't)